### DISCUSSION TITLE
download_models.sh tells me it could not select a nameless driver with gpu capabilities

### DISCUSSION THREAD

**Discussion by obones** [toverainc/willow-inference-server#70] [Q&A]
Hello,

As I'm following the instructions on an Arch Linux server, I have reached the `download_models.sh` part of things and I receive the following error:

```
Using default configuration values
docker: Error response from daemon: could not select device driver "" with capabilities: [[gpu]].
```

Now, I have the following hardware:

Intel HD Graphics 530 (i915 driver)
Intel(R) Core(TM) i7-6700 CPU @ 3.40GHz

It has served me well until now because there are no special graphics requirements on that machine.
Would there be a way to have this project run on this machine?

If not, I have both a GT710 and GT610 lying around here (with 1G RAM), but from what I gather, they would not help much with this situation?

Thanks for any pointers

**Comment by kristiankielhofner** [comment]
None of those GPUs are supported. iGPU support just isn't fundamentally possible and your Nvidia devices are too old (nine years) to remotely run WIS. For WIS-only GPU applications I recommend at least the Tesla P4 or GTX 1070, which can be purchased from eBay or other resellers for $90 - $130 or so. Of course this will vary with your country but one of the Willow developers recently purchased a Tesla P4 in Belgium for 120 EUR or something like that.

We are working on refining CPU only support for WIS but I cannot stress how enough how different of an experience it will be on that CPU. You will likely be waiting at least several seconds for speech recognition and command action. The GPUs I mentioned are at least an order of magnitude faster and can do it in 500ms.

**Reply by obones** [reply]
Thanks, that's what I thought.
Adding such a GPU card inside the current server will prove a challenge, especially considering it requires two case slots when they are already all used.

**Reply by kristiankielhofner** [reply]
The Tesla P4 is very popular with Willow. It's single slot, roughly $100, and doesn't require any additional power or cooling.

**Reply by lordratner** [reply]
What about the Tesla P40? It has 24GB of VRAM. Twice the price, but 3 times the RAM. Is this card compatible with Willow, and if so, will the 24GB of RAM allow for the same functionality as the RTX 3090, as far as running the LLMs? I expect it to be much slower, but I believe it still runs faster than a GTX 1070.

Thoughts?

**Reply by kristiankielhofner** [reply]
I don't have experience with that model but what I can guess from being familiar with the architectures is that it will likely be incredibly slow for an LLM. The strange thing about that variant (now) is that there aren't really many practical applications (off the top of my head) that could actually make use of the extra VRAM in a worthwhile way.

**Reply by lordratner** [reply]
Ok, perfect, thanks!
